crates/cli/src/config_file.rs

```diff
@@ -22,26 +20,117 @@ pub fn read_config_file(path: &str) -> Result<Option<ConfigFile>> {
     ));
 
     // first, static-analysis.datadog.yml
-    let mut file = match File::open(yml_file_path) {
-        Ok(f) => f,
+    match File::open(yml_file_path) {
+        Ok(f) => Ok(Some(f)),
         Err(e1) if e1.kind() == std::io::ErrorKind::NotFound => {
             // second, static-analysis.datadog.yaml
             match File::open(yaml_file_path) {
-                Ok(f) => f,
-                Err(e2) if e2.kind() == std::io::ErrorKind::NotFound => return Ok(None),
-                otherwise => otherwise?,
+                Ok(f) => Ok(Some(f)),
+                Err(e2) if e2.kind() == std::io::ErrorKind::NotFound => Ok(None),
+                _ => Err(anyhow!("cannot open config file")),
             }
         }
-        otherwise => otherwise?,
-    };
-    let mut contents = String::new();
-
-    let size_read = file
-        .read_to_string(&mut contents)
-        .context("error when reading the configration file")?;
-    if size_read == 0 {
-        return Err(anyhow!("the config file is empty"));
+        _ => Err(anyhow!("cannot open config file")),
+    }
+}
+
+// We first try to read static-analysis.datadog.yml
+// If it fails, we try to read static-analysis.datadog.yaml
+// If the file does not exist, we return a Ok(None).
+// If there is an error reading the file, we return a failure
+pub fn read_config_file(path: &str) -> Result<Option<ConfigFile>> {
+    match get_config_file(path) {
+        Ok(file_opt) => match file_opt {
+            Some(mut file) => {
+                let mut contents = String::new();
+
+                let size_read = file
+                    .read_to_string(&mut contents)
+                    .context("error when reading the configration file")?;
+                if size_read == 0 {
+                    return Err(anyhow!("the config file is empty"));
+                }
+                match parse_config_file(&contents) {
+                    Ok(c) => Ok(Some(c)),
+                    Err(e) => Err(e),
+                }
+            }
+            None => Ok(None),
+        },
+        Err(e) => Err(e),
```

Comment:
the Err case in the outer match just returns the error, so why not just use `?` to propagate the error if any, and use a `let ... Some` to bail early when it's None as well: 
```rs
let Some(mut file) = get_config_file(path)? else {
    return Ok(None);
};
```